src/Form/PasswordChange.php

```diff
@@ -0,0 +1,163 @@
+<?php
+
+namespace Drupal\password_enhancements\Form;
+
+use Drupal\Component\Datetime\TimeInterface;
+use Drupal\Component\Utility\Crypt;
+use Drupal\Core\Form\FormBase;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\Password\PasswordInterface;
+use Drupal\Core\Session\SessionManagerInterface;
+use Drupal\user\UserInterface;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Password change form.
+ */
+class PasswordChange extends FormBase {
+
+  /**
+   * Date time.
+   *
+   * @var \Drupal\Component\Datetime\TimeInterface
+   */
+  protected $dateTime;
+
+  /**
+   * Current user.
+   *
+   * @var \Drupal\Core\Session\AccountProxyInterface
+   */
+  protected $user;
```

Comment:
This is a `\Drupal\user\UserInterface` if you load the complete user object from entity storage.